rust/src/lib/ifaces/inter_ifaces.rs

```diff
@@ -1017,6 +1019,117 @@ impl MergedInterfaces {
         Ok(())
     }
 
+    fn get_hsr_mac(&self, iface: &HsrInterface) -> Option<String> {
```

Comment:
This is a function dedicate for HSR, not a function shared by multiple interface types, so please move this function to `ifaces/hsr.rs` like:

```rust
impl HsrInterface {

fn get_hsr_mac(&self, merged_ifaces: MergedInterfaces) {
}
}
```

}